File: algorithms/easy/keep_multiplying_found_values_by_two.ex
defmodule Solution do
  @spec find_final_value(nums :: [integer], original :: integer) :: integer
  def find_final_value(nums, original) do
    set = MapSet.new(nums)

    Stream.iterate(original, &(&1 * 2))
    |> Stream.drop_while(&MapSet.member?(set, &1))
    |> Enum.take(1)
    |> hd()
  end
end

File: algorithms/easy/keep_multiplying_found_values_by_two.rs
use std::collections::HashSet;
use std::iter::FromIterator;

impl Solution {
    pub fn find_final_value(nums: Vec<i32>, mut original: i32) -> i32 {
        let set: HashSet<i32> = HashSet::from_iter(nums);
        while set.contains(&original) {
            original *= 2;
        }
        original
    }
}
